From: Jiapeng Zhong <abaci-bugfix@linux.alibaba.com>
Date: Tue, 26 Jan 2021 16:42:34 +0800
Subject: scsi: qla2xxx: Simplify the calculation of variables
Patch-mainline: v5.12-rc1
Git-commit: 99de0ea06ac04c084b42da3a15aefd4bde1c8424
References: bsc#1185491

Fix the following coccicheck warnings:

./drivers/scsi/qla2xxx/qla_nvme.c:288:24-26: WARNING !A || A && B is
equivalent to !A || B.

Link: https://lore.kernel.org/r/1611650554-33019-1-git-send-email-abaci-bugfix@linux.alibaba.com
Reported-by: Abaci Robot <abaci@linux.alibaba.com>
Reviewed-by: Himanshu Madhani <himanshu.madhani@oracle.com>
Signed-off-by: Jiapeng Zhong <abaci-bugfix@linux.alibaba.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Daniel Wagner <dwagner@suse.de>
---
 drivers/scsi/qla2xxx/qla_nvme.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/scsi/qla2xxx/qla_nvme.c
+++ b/drivers/scsi/qla2xxx/qla_nvme.c
@@ -292,7 +292,7 @@ static int qla_nvme_ls_req(struct nvme_f
 	struct qla_hw_data *ha;
 	srb_t           *sp;
 
-	if (!fcport || (fcport && fcport->deleted))
+	if (!fcport || fcport->deleted)
 		return rval;
 
 	vha = fcport->vha;
